DESCRIPTION.md

```diff
@@ -8,6 +8,10 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
 
 # Release Notes
 
+- v3.10.1(May 21, 2024)
+
+  - Removed incorrect error log message in arrow data conversion.
```

Comment:
- Removed an incorrect error log message that could occur during arrow data conversion.